feat(trace): correct loading and add better error state

JonasBa opened the pull request:
rename GetTraceType to shape as it is more descriptive and more accurate of what it does (the shape can also change through the course as we fetch more data). 

Adds a proper loading and error state based on the request status

JonasBa committed:
ref(trace): rename GetTraceType to shape

JonasBa committed:
feat(trace): add proper loading state

JonasBa committed:
feat(trace): skip rerendering indicator if not changed

evanpurkhiser commented on static/app/components/feedback/widget/useFeedbackWidget.tsx:
```diff
@@ -7,7 +7,7 @@ import ConfigStore from 'sentry/stores/configStore';
 
 interface Props {
-  buttonRef?: RefObject<HTMLButtonElement>;
+  buttonRef?: RefObject<HTMLButtonElement> | RefObject<HTMLAnchorElement>;
```
```suggestion
  buttonRef?: RefObject<HTMLButtonElement | HTMLAnchorElement>;
```

can't it be written like this

evanpurkhiser replied:
Oh really? What's the difference in this scenario?

JonasBa replied:
Yeah, correct. Hopefully this [explains](https://www.typescriptlang.org/play?#code/JYWwDg9gTgLgBDAnmApnA3gJRQMwPIBGAVigMYwC+cOUEIcARFCgIbkMBQHOArgHblgEPnADuwACYoAFMxwAuONnzEyMADwAJACoBZADIAhHjBjCAogBsUIFHxgAfHQYCCAgBbQrNuzAB8AJToFNz8gsJwfCxQtKKyuIrKhCTkWnpGJmZ83rb2fnAOSrjJammuHl7Wuf5BIRykwgDO8OJSAPpyicWqqc4ZphZVvk7pbqSeUDm++QC8GHCkPDG+inw8lpYhDXzNkdGxHQlFKikafcYD2UN5cHPoC0vM9qvrm1ytMh+HOAEcUTEQOJfOS-f6xaRgwHfX5cIA) it

evanpurkhiser replied:
Yeah I see what you're saying. Subtle. 👍

JonasBa committed:
feat(trace): use react query for fetching trace data (#66664)

React query caches the response so if users navigate to this page while
they do perf investigation, the data will be pulled from cache and
instantly populated.

@Abdkhan14, we should do the same for the meta query, right now those
top level user/browser and duration fields are refetched on each
navigation, which isnt ideal and we should change that, but it's a good
start as the majority of the data will at least be here.

JonasBa committed:
Merge branch 'master' into jb/trace/error-state

getsantry[bot] committed:
:hammer_and_wrench: apply pre-commit fixes
